qa/tasks/cephfs/filesystem.py

```diff
@@ -611,7 +583,41 @@ def create(self):
 
         self.getinfo(refresh = True)
 
-        
+    def destroy(self, reset_obj_attrs=True):
+        log.info('Destroying file system ' + self.name +  ' and related '
+                 'pools')
+
+        # make sure no MDSs are attached to given FS.
+        self.mon_manager.raw_cluster_cmd('fs', 'fail', self.name)
+        self.mon_manager.raw_cluster_cmd(
+            'fs', 'rm', self.name, '--yes-i-really-mean-it')
+
+        self.mon_manager.raw_cluster_cmd('osd', 'pool', 'rm',
+            self.get_metadata_pool_name(), self.get_metadata_pool_name(),
+            '--yes-i-really-really-mean-it')
+        for poolname in self.get_data_pool_names():
+            try:
+                self.mon_manager.raw_cluster_cmd('osd', 'pool', 'rm', poolname,
+                    poolname, '--yes-i-really-really-mean-it')
+            except CommandFailedError as e:
+                if e.exitstatus == 16: # EBUSY, this data pool is used
```

Comment:
I did not touch this block of code intentionally since my requirement was only to indent it with one more level. I've `s/16/EBUSY` locally, I'll update the branch after tests.

Reply:
Ah OK!